app/controllers/lists_controller.rb

```diff
@@ -21,6 +60,29 @@ def find_board
     redirect_to root_path
   end
 
+  def find_list
+    @list = List.find_by id: params[:list_id]
+    return if @list
+
+    flash[:danger] = t ".cant_find"
+    redirect_to root_path
+  end
+
+  def update_list_position old_position, new_position, board, changed_list_id
+    lists = board.lists
+    if old_position < new_position
+      lists.greater(old_position)
+           .less(new_position + Settings.data.confirm)
+           .not_id(changed_list_id)
+           .update_all("position = position - 1")
+    elsif old_position > new_position
+      lists.greater(new_position - Settings.data.confirm)
+           .less(old_position)
+           .not_id(changed_list_id)
+           .update_all("position = position + 1")
```

Comment:
đoạn code này khá giống nhau, refactor đoạn này đi nhé

Reply:
vâng

Reply:
vâng nhìn nó giống nhưng mà 2 trường hợp lại cần truyền params khác nhau lắm anh cái thuật toán di chuyển nó thế ạ, em sẽ refactor lại cái đoạn update_all thêm transaction ạ